src/lib/util/buyLimit.ts

```diff
@@ -0,0 +1,68 @@
+import { Bank } from 'oldschooljs';
+import { Item } from 'oldschooljs/dist/meta/types';
+
+export function buyLimit({
+	buyLimitBank,
+	increaseFactor,
+	itemBeingBought,
+	quantityBeingBought,
+	baseCost,
+	absoluteLimit
+}: {
+	buyLimitBank: Bank;
+	/**
+	 * The price multiplier increases by 1 for each multiple of this amount they have bought. (1x, 2x, 3x, etc)
+	 */
+	increaseFactor: number;
+	/**
+	 * You cannot buy any more than this.
+	 */
+	absoluteLimit: number;
+	itemBeingBought: number | Item;
+	quantityBeingBought: number;
+	baseCost: number;
+}) {
+	const itemID = typeof itemBeingBought === 'number' ? itemBeingBought : itemBeingBought.id;
+	const amountBought = buyLimitBank.amount(itemID);
+
+	let amountToBuy = 0;
+	let finalCost = 0;
+
+	if (quantityBeingBought <= 0) {
+		return { amountToBuy, finalCost };
+	}
+
+	const maxCanBuy = absoluteLimit - amountBought;
+	const allowedBuyAmount = quantityBeingBought > maxCanBuy ? maxCanBuy : quantityBeingBought;
+	// CHANGE THIS TO MATCH currentMultiplier
+	// let currentMultiplier = Math.floor(amountBought / increaseFactor) + 1;
```

Comment:
this code line can be removed, but the comment above it might be useful to keep for  any possible algorithm changes in the future